stl/inc/yvals_core.h

```diff
@@ -1515,6 +1511,14 @@
 #endif // language mode
 #endif // _M_CEE
 
+#if defined(__cpp_lib_concepts)
+#if _HAS_CXX23
+#define __cpp_lib_ranges 202207L // P2494R2 Relaxing Range Adaptors to Allow for Move-only Types
+#elif _HAS_CXX20
+#define __cpp_lib_ranges 202110L // P2415R2 What is a view?
+#endif
+#endif // defined(__cpp_lib_concepts)

```

Comment:
Currently `__cpp_lib_ranges` has different values in C++20 and C++23 mode, but the change in `<ranges>` are all applied to C++20 mode ("Yet another DR against C++20"?).

Perhaps we should choose either
- treat P2494R2 as a DR and just bump `__cpp_lib_ranges` to `202207L` in C++20 mode, or
- only apply the change to C++23 mode and differently define `__cpp_lib_ranges`.

In the latter case, we can consistently use `_Adoptable_object` and `_Adoptable_box` in the definition of range adaptors, while `_Adoptable_object` and `_Adoptable_box` (only the constraints, not the class body) are differently defined in C++20 and C++23 modes.

And `// TRANSITION, GH-395` should be kept IMO.